### ISSUE TITLE
Add support for debounce

### ISSUE THREAD

**Issue by seanmedcalf** [trevoreyre/autocomplete#60]
I've tried to add debouncing to the search function but the component will not update after the results array is updated. I also tried to call the handleUpdate and handleShow methods directly which will show the results appropriately but clicking on any result fails to populate the event with the result object. Is there any way to update the control after a debounced API call completes? Thanks.

**Comment by versedi** [comment]
I'm using it with debounce.

You have to create your own vanilla js component cause you need to add a few lines inside  `initialize` method which afaik can't be extended otherwise. 


```
//oold code
        this.input.setAttribute('aria-owns', this.resultList.id)

        document.body.addEventListener('click', this.handleDocumentClick)

//new part
        let inputHandler = event => {
            const { value } = event.target
            console.log('value: ' + value);
            this.core.updateResults(value);
            this.value = value
        }

        let debouncedInputHandler = debounce(inputHandler, 200);
        this.input.addEventListener('input', debouncedInputHandler);

//old part continues
(...)
```

**Comment by apisklov** [comment]
Hi.
How to use debounce in Vue component?
`this.search = _.debounce(this.getAutocomplete, 500)`
This code does not update the results.

**Comment by trevoreyre** [comment]
@seanmedcalf I agree there should be easier support for debouncing. I don't know if I necessarily would want to build it in to the component, but it should at least be easier to override the default handling of the input event so you could debounce that yourself.

Do you happen to have a code sample of what you've tried for debouncing so far?

**Comment by seanmedcalf** [comment]
Holidays over, time for me to get back to work...

My first attempt was to use the debounce approach referenced in this article: https://levelup.gitconnected.com/debounce-in-javascript-improve-your-applications-performance-5b01855e086

```
<autocomplete
      ref="autocomplete"
      :search="debounce(search, 350)"
      :placeholder="$t('user-search.placeholder')"
      :aria-label="$t('user-search.placeholder')"
      auto-select
      :default-value="selectedValue"
      :get-result-value="getUserFullName"
    >
```

Unfortunately it was causing the following error:
```
autocomplete.esm.js:232 Uncaught (in promise) TypeError: Cannot read property 'length' of undefined
    at autocomplete.esm.js:232
```
Unless you have any simpler suggestions, I can look deeper into the approach @versedi took.

**Comment by jssouders** [comment]
@trevoreyre @seanmedcalf I added the debounce functionality to the Vue component along with updated documentation on how to use it.  The changes are in the following PR
**PR:** [VUE Debounce Functionality](https://github.com/trevoreyre/autocomplete/pull/66/files)

@versedi If you want to take a look at what I implemented in Vue component you should be able to add the same functionality in the Javascript component.

**Comment by trevoreyre** [comment]
Added a `debounceTime` prop to both JS and Vue components. It’s available now in v2.1.0.
